subprojects/enterprise/src/main/java/org/gradle/internal/enterprise/impl/DefaultGradleEnterprisePluginBackgroundJobExecutor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.enterprise.impl;
+
+import org.gradle.internal.classpath.Instrumented;
+import org.gradle.internal.concurrent.ExecutorPolicy;
+import org.gradle.internal.concurrent.ManagedExecutor;
+import org.gradle.internal.concurrent.ManagedExecutorImpl;
+import org.gradle.internal.enterprise.GradleEnterprisePluginBackgroundJobExecutor;
+import org.gradle.internal.service.scopes.Scopes;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+@ServiceScope(Scopes.Gradle.class)
+public class DefaultGradleEnterprisePluginBackgroundJobExecutor implements GradleEnterprisePluginBackgroundJobExecutor {
+    private final ManagedExecutor executorService = createExecutor();
+
+    private static ManagedExecutor createExecutor() {
+        ThreadPoolExecutor poolExecutor = new ThreadPoolExecutor(
+            4, 4,
+            30, TimeUnit.SECONDS,
+            new LinkedBlockingQueue<>(),
+            new BackgroundThreadFactory()
+        );
+        poolExecutor.allowCoreThreadTimeOut(true);
+
+        return new ManagedExecutorImpl(poolExecutor, new ExecutorPolicy.CatchAndRecordFailures());
+    }
+
+    @Override
+    public void execute(@Nonnull Runnable job) {
+        executorService.execute(() -> executeWithoutConfigurationInputTracking(job));
+    }
+
+    private static void executeWithoutConfigurationInputTracking(Runnable job) {
```

Comment:
Not a strong feeling but I'd move this function to `Instrumented` and hide the individual `disable` / `restore` steps completely.